### PR TITLE
Use fake version instead of 999-SNAPSHOT in devtools-testing

### PR DESCRIPTION
**Pull request by ia3andy**
Following suggestion from https://github.com/quarkusio/quarkus/pull/16692#pullrequestreview-643104739

### CODE REVIEW

File: devtools/project-core-extension-codestarts/src/test/java/io/quarkus/devtools/codestarts/quarkus/ConfigYamlCodestartTest.java [hunk 1]
```diff
@@ -20,7 +21,12 @@
     void testContent() throws Throwable {
         codestartTest.checkGeneratedSource("org.acme.GreetingConfig");
         codestartTest.assertThatGeneratedFileMatchSnapshot(JAVA, "src/main/resources/application.yml");
-        //codestartTest.buildAllProjects();
+    }
+
+    @Test
+    @EnabledIfSystemProperty(named = "build-projects", matches = "true")
```
Comment (ia3andy) [inline devtools/project-core-extension-codestarts/src/test/java/io/quarkus/devtools/codestarts/quarkus/ConfigYamlCodestartTest.java:27]: @glefloch better this way right :) ?

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/utils/NestedMaps.java [hunk 2]
```diff
@@ -57,11 +64,13 @@ public static void deepMerge(Map left, Map right) {
                     || rightValue instanceof Float
                     || rightValue instanceof Long
                     || rightValue instanceof Double
-                    || rightValue instanceof String) {
+                    || rightValue instanceof String
+                    || rightValue == null) {
                 // Override
                 left.put(key, rightValue);
+            } else {
+                throw new IllegalArgumentException("Invalid value type for deepMerge: " + rightValue.getClass());
```
Comment (ia3andy) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/utils/NestedMaps.java:72]: @aloubyansky as discussed, it now throws an error if an unexpected object is used as input. I also added a safety to never modify the left map by adding a hat method.

**Review by aloubyansky — APPROVED** [review]

File: devtools/project-core-extension-codestarts/src/test/java/io/quarkus/devtools/codestarts/quarkus/ConfigYamlCodestartTest.java (hunk 1 above)
Comment (glefloch) [reply devtools/project-core-extension-codestarts/src/test/java/io/quarkus/devtools/codestarts/quarkus/ConfigYamlCodestartTest.java:27]: right :)

**Comment by ia3andy** [comment]
This doesn't seem to be enough to fix the snapshots, we also need to use the mocked platform in the core codestarts.. I will provide another PR for this..

**Comment by ia3andy** [comment]
> This doesn't seem to be enough to fix the snapshots, we also need to use the mocked platform in the core codestarts.. I will provide another PR for this..

It's in this PR in another commit

**Pull request merged**